src/main/kotlin/no/nav/tilleggsstonader/sak/satsjustering/SatsjusteringService.kt

```diff
@@ -0,0 +1,116 @@
+             * så kan man kaste RekjørSenereException her, og tasken er ansvarlig for å kjøre på nytt dagen etter.
+             */
+            throw RekjørSenereException(
+                "Fagsak=$fagsakId har en ikke ferdigstilt behandling, kan ikke kjøre satsjustering.",
+                LocalDate.now().plusDays(1).atTime(7, 0),
+            )
+        }
+        val sisteIverksatteBehandling = behandlingService.finnSisteIverksatteBehandling(fagsakId)
+        if (sisteIverksatteBehandling?.id != behandlingId) {
+            logger.info(
+                "Siste iverksatte behandling=${sisteIverksatteBehandling?.id} er ikke lik behandlingId=$behandlingId, kan ikke kjøre satsjustering.",
+            )
+            return
+        }
+
+        opprettRevurderingOgKjørSatsendring(fagsakId)
+    }
+
+    private fun opprettRevurderingOgKjørSatsendring(fagsakId: FagsakId) {
+        val revurdering = opprettRevurderingForSatsendring(fagsakId)
+
+        val forrigeIverksatteBehandlingId =
+            revurdering.forrigeIverksatteBehandlingId
+                ?: error("Revurdering for fagsak=${revurdering.fagsakId} har ikke fått forrigeIverksattBehandlingId")
+        val vedtaksperioder =
+            vedtakservice.hentVedtaksperioder(behandlingId = forrigeIverksatteBehandlingId).tilDto()
+
+        beregnYtelseSteg.lagreVedtakForSatsjustering(
+            saksbehandling = revurdering,
+            vedtak = InnvilgelseLæremidlerRequest(vedtaksperioder = vedtaksperioder),
+            satsjusteringFra = finnDatoForSatsjustering(revurdering),
+        )
+
+        // Kaller disse direkte for å hoppe over totrinnskontroll
+        behandlingService.oppdaterStatusPåBehandling(revurdering.id, BehandlingStatus.FATTER_VEDTAK)
+        behandlingService.oppdaterResultatPåBehandling(revurdering.id, BehandlingResultat.INNVILGET)
+        behandlingService.oppdaterStegPåBehandling(revurdering.id, StegType.BESLUTTE_VEDTAK)
+        iverksettService.iverksettBehandlingFørsteGang(revurdering.id)
+
+        // Her lages også internt vedtak og behandling- og vedtaksstatistikk
+        ferdigstillBehandlingSteg.utførSteg(revurdering, null)
+    }
+
+    private fun opprettRevurderingForSatsendring(fagsakId: FagsakId): Saksbehandling {
+        val revurderingId =
+            revurderingBehandlingService.opprettBehandling(
+                OpprettRevurdering(
+                    fagsakId = fagsakId,
+                    årsak = BehandlingÅrsak.SATSENDRING,
+                    valgteBarn = emptySet(), // TODO - kopier over barn
+                    nyeOpplysningerMetadata = null,
+                    kravMottatt = null,
+                    skalOppretteOppgave = false,
+                ),
+            )
+        // Gjenbruker grunnlag fra forrige behandling ved satsjustering
+        faktaGrunnlagService.kopierGrunnlagsdataFraForrigeIverksatteBehandling(revurderingId)
```

Comment:
Om behandlingen ikke har noe grunnlagsdata feiler det ved visning frontend. Tar derfor å kopierer over grunnlaget fra forrige iverksatte, da vi ved en satsjustering ikke tar høyde for nye eventuelle opplysninger som skal ha kommet. I en "vanlig" revurdering ville grunnlagsdataen blitt hentet når man åpnet behandlingen første gang